test/lsptoolshost/integrationTests/integrationHelpers.ts

```diff
@@ -50,7 +62,7 @@ export async function activateCSharpExtension(): Promise<void> {
     console.log('ms-dotnettools.csharp activated');
     console.log(`Extension Log Directory: ${csharpExtension.exports.logDirectory}`);
 
-    if (shouldRestart) {
+    if (shouldRestart && allowRestart) {
```

Comment:
this has a good chance of being flaky.  Without restarting after a restore we have no consistent project load state in the test.  And other tests may have changed various settings that impact this running test.

C# devkit has a similar issue (where we cannot restart) - but it is resolved by running each suite (file) in a separate test run *and* devkit project load ensures that all restores and everything is complete.  We could consider running the razor tests only in the devkit mode.

Reply:
Yea I had a feeling that was the answer. I have to head out but I'll reach out tomorrow and get something more solid here. It's possible we can gracefully handle a restart in razor with some extra work.

Reply:
I had similar thoughts. I thought the restart was there because we might have had to download the .NET runtime or something, so could see that not working in CI for sure.

I'm curious what state do we need between tests in Razor, that this causes problems?

Reply:
Talked offline, we can use DevKit tests to avoid restart